app/src/main/java/org/projectbuendia/client/net/OdkDatabase.java

```diff
@@ -36,18 +36,21 @@ public static long getFormIdForPath(File path) {
                 path, new String[] {
                     BaseColumns._ID
                 });
-            // There should only ever be one form per UUID. But if something goes wrong, we want the
-            // app to keep working. Assume the latest one is correct.
-            if (cursor.getCount() > 1) {
-                Log.e(TAG, "More than one form in database with the same id. This indicates an "
-                    + "error occurred on insert (probably a race condition) and should be "
-                    + "fixed. However, the app should still function correctly");
+            //if any form was found by the given file.
+            if(cursor.getCount() >= 1) {
+                // There should only ever be one form per UUID. But if something goes wrong, we want
+                // the app to keep working. Assume the latest one is correct.
+                if (cursor.getCount() > 1) {
+                    Log.e(TAG, "More than one form in database with the same id. This indicates an "
+                        + "error occurred on insert (probably a race condition) and should be "
+                        + "fixed. However, the app should still function correctly");
+                }
+                Preconditions.checkArgument(cursor.getColumnCount() == 1);
```

Comment:
This check is redundant; the projection in the query is just 10 lines up and there's only one column.